demisto_sdk/commands/common/hook_validations/id.py

```diff
@@ -289,31 +289,37 @@ def is_entity_version_match_playbook_version(self, implemented_entity_list_from_
         Returns:
             bool. Whether the playbook's version match playbook's entities.
         """
+        invalid_version_entities = []
+        is_valid = True
         for entity_data_dict in entity_set_from_id_set:
-            if not implemented_entity_list_from_playbook:
-                return True
 
             entity_id = list(entity_data_dict.keys())[0]
             all_entity_fields = entity_data_dict[entity_id]
             entity_name = entity_id if entity_id in implemented_entity_list_from_playbook else all_entity_fields.get(
                 "name")
-
             is_entity_used_in_playbook = entity_name in implemented_entity_list_from_playbook
 
             if is_entity_used_in_playbook:
                 entity_version = all_entity_fields.get("fromversion", "")
                 is_version_valid = not entity_version or LooseVersion(entity_version) <= LooseVersion(
                     main_playbook_version)
-                if is_version_valid:
-                    implemented_entity_list_from_playbook.remove(entity_name)
+                if not is_version_valid:
+                    invalid_version_entities.append(entity_name)
+                implemented_entity_list_from_playbook.remove(entity_name)
```

Comment:
better to copy the list before as this is causing a side effect of clearing the list (it is past by reference to the function) and might affect later logic in the outer scopes.